web/src/components/Upload/util.ts

```diff
@@ -1,43 +1,132 @@
 import { FileWithPath } from 'react-dropzone'
+import moment from 'moment-timezone'
+import { DustrakMeta, ValidateMeta } from 'components/Upload/types'
 
-// export const parseDatesFromCsv = (file: FileWithPath): Array<string> => {}
+export const getGpsStart = (textLines: Array<string>): moment.Moment => {
+  for (const line of textLines) {
+    if (line.startsWith('$GPRMC')) {
+      const lineFields: Array<string> = line.split(',', 10)
+      const encodedTime: string = lineFields[1]
+      const encodedDate: string = lineFields[9]
+      const endTime: moment.Moment = moment.utc(
+        `${encodedDate} ${encodedTime}`,
+        'DDMMYYYY hh mm ss.SS'
+      )
+      return endTime
+    }
+  }
+  return moment.utc('')
+}
 
-const parseDateFromLogFileName = (fileName: string): string => {
-  const fileNameArr = fileName.split('_')
-  const dateString = fileNameArr[1]
-  return `${dateString.slice(0, 4)}/${dateString.slice(
-    4,
-    6
-  )}/${dateString.slice(6)}`
+export const getDustrakStart = (textLines: Array<string>): moment.Moment => {
+  try {
+    const startTime: string = textLines[6].split(',')[1]
+    const startDate: string = textLines[7].split(',')[1]
+    const startDatetime: moment.Moment = moment.tz(
+      `${startDate} ${startTime}`,
+      'MM-DD-YYYY hh:mm:ss a',
+      'America/Los_Angeles'
+    )
+    return startDatetime
+  } catch (e) {
+    if (e instanceof TypeError) {
```

Comment:
can you add a comment on when/why a `TypeError` occurs here and why we want to return a moment object in that scenario?

Reply:
This was originally an `if` statement covering the scenario where the file text array isn't long enough to have values in that array position. It's a type error because the index out of range condition returns `undefined` and you can't `split type of undefined`. Rather than comment, I think it's better practice to cover the known error condition with an `if` statement- it's more expressive in the way it's written and prevents excessive errors being triggered. I'll push an update to `if`. But I'll welcome your thoughts on whether it's better to keep it as an error.